src/Microsoft.AspNet.Identity/SignInManager.cs

```diff
@@ -406,8 +405,12 @@ public virtual AuthenticationProperties ConfigureExternalAuthenticationPropertie
         /// <returns></returns>
         protected async virtual Task<bool> LogResultAsync(bool result, TUser user, [System.Runtime.CompilerServices.CallerMemberName] string methodName = "")
         {
-            Logger.WriteInformation(Resources.FormatLoggingSigninResult(Resources.FormatLoggingResultMessage(methodName,
-                await UserManager.GetUserIdAsync(user)), result));
+            // Check if log level is enabled before creating the message.
```

Comment:
This sort of goes back to my point about the result object knowing how to render its log message, as today you'd have to put a switch inside SignInManager's Log to do this, while I think this decision should be taken care of by the result object (whether its currently a success or a failure).  That was why I origionally suggested for the Log to happen inside the result, while the managers method was mostly just to format the outer string and pass it into XXXResult.Log(outerformat), and the result call the appropriate Log method with warning/information since only the result knows truly whether its a failure or success